src/coreclr/jit/rangecheck.cpp

```diff
@@ -1033,6 +1067,11 @@ Range RangeCheck::ComputeRangeForLocalDef(BasicBlock*          block,
     LclSsaVarDsc* ssaDef = GetSsaDefAsg(lcl);
     if (ssaDef == nullptr)
     {
+        LclVarDsc* varDsc = m_pCompiler->lvaGetDesc(lcl->GetLclNum());
+        if (varDsc->lvIsParam && (lcl->GetSsaNum() == SsaConfig::FIRST_SSA_NUM))
+        {
+            return GetRangeFromType(varDsc->TypeGet());
+        }
```

Comment:
Heh, parameters are `NormalizeOnLoad`, so the small type for them doesn't mean anything (they can have any `TYP_INT` value, unless normalized by a preceding store, but here the check is very specifically for the out-of-method definition).

Is it actually needed?